Administrator/src/main/java/com/SS/Administrator/Controller/AuthorController.java

```diff
@@ -21,104 +22,79 @@
 
 
 @RestController
+@RequestMapping(path = "/admin/authors")
 public class AuthorController {
 
 	@Autowired
-	AuthorService authorService;
+	private AuthorService authorService;
 	
-	@RequestMapping(path = "/admin/authors", method = RequestMethod.GET)
+	@RequestMapping(method = RequestMethod.GET,produces ={
+			MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
 	public ResponseEntity<List<Author>> getAuthors(){
 		List <Author> authors = authorService.readAllAuthors();
 		return new ResponseEntity<List<Author>>(authors,HttpStatus.OK);
 	}
 	
-	@RequestMapping(path = "/admin/authors", method = {RequestMethod.DELETE, RequestMethod.PUT})
-	public ResponseEntity<String> authorMethodNotAllowed() {
-		return new ResponseEntity<String>(HttpStatus.METHOD_NOT_ALLOWED);
+	@RequestMapping(path = "/{authorId}", method = RequestMethod.GET,produces ={
+			MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
+	public ResponseEntity<Author> getAuthor(@PathVariable@NotBlank int authorId){
+		Author author = authorService.findAuthorById(authorId);
+		return new ResponseEntity<Author>(author,HttpStatus.OK);
 	}
 	
-	@RequestMapping(path = "/admin/authors/{authorId}", method = RequestMethod.GET)
-	public ResponseEntity<Author> getAuthor(@PathVariable int authorId){
-		try {
-			Author author = authorService.findAuthorById(authorId);
-			return new ResponseEntity<Author>(author,HttpStatus.OK);
-		}catch(NoSuchElementException e) {
-			return new ResponseEntity<Author>(HttpStatus.NOT_FOUND);
-		}
-	}
-	@RequestMapping(path = {"/admin/authors"} , method = RequestMethod.POST)
+	@RequestMapping(method = RequestMethod.POST, consumes ={MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE },
+			produces ={MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
 	public ResponseEntity<Author> addAuthor(@RequestBody Author author){

```

Comment:
Add @Validate in front of @RequestBody Annotation. @NotNull and @NotEmpty should be used on Entity class.